### ISSUE TITLE
Consider adding heirarchal comments

### ISSUE THREAD

**Issue by nickserv** [toroidal-code/codeflash#194]
Since discussions on Codeflash would often involve multiple issues/suggestions about one solution, I feel that comments could quickly get very out of hand if we force them to be linear. On Stack Overflow, you're usually not going to have a large number of different issues about one answer being discussed in comments (without the question being downvoted off the page), so heirarchal comments aren't really needed. On Reddit, people often discuss opinions on a post in more detail, and elaborate on why they agree/disagree with each other. I feel that Codeflash would follow a somewhat similar model of communiation, since it will have a strong focus on letting different people judge how well code solves a problem together. As a result, I feel that comment threads would get too involved to be easily readable/manageable in a totally linear fashion.

We can follow forums and let users quote other users, or follow Twitter/GitHub and let users reply directly to each other by username. However, I think Reddit's solution for this works really well. Whenever a user replies to another comment, his/her new comment is created as a subcomment of the original comment, with indents to represent the hierarchy (just like source code!).

**tl;dr** Consider making comments work like they do on Reddit.

**Comment by kristenmills** [comment]
I thought we already had this discussion already and immediately nixed it.

**Comment by StacyJCarlos** [comment]
I was for linear commenting at first, but editing an entire solution I think would take more than a one comment. Maybe a comment hierarchy would be a bit much for large comments, but have a way to reference other users would be nice.

**Comment by smy2748** [comment]
The issue with Reddit's nested comments is that they quickly become expanding chains that people try to follow all the way "down the rabbit-hole" and comments that are on the top level may never get read because the users only read through the chain of the first couple comments they see.

Quoting other users is nice especially since it preserves what a user previously said so that even if they edit it later, the comment is understandable.

**Comment by nickserv** [comment]
@smy2748 That's a good point. Reddit is a lot more conversation oriented than we will be anyway.

**Comment by nickserv** [comment]
Coding Horror has a really nice article on flat vs. threaded conversations: http://www.codinghorror.com/blog/2012/12/web-discussions-flat-by-design.html

I'm starting to think that hierarchical comments would probably be too complicated for Codeflash. Should I close this with a no?

**Comment by kristenmills** [comment]
correct
